encodings/sparse/benches/canonical.rs

```diff
@@ -0,0 +1,49 @@
+#![allow(clippy::unwrap_used)]
+
+use divan::Bencher;
+use vortex_array::builders::{ArrayBuilder, PrimitiveBuilder};
+use vortex_array::{IntoArray, IntoCanonical};
+use vortex_buffer::BufferMut;
+use vortex_dtype::Nullability;
+use vortex_scalar::Scalar;
+use vortex_sparse::SparseArray;
+
+fn generate_sparse_array(len: usize, sparsity: f64) -> SparseArray {
```

Comment:
tiny style nits: 
- could consider going ordering functions by call order (been told by @gatesn that's how we do it :) 
=> main first, then the benchmarks, then the helper. I primarily care about the benchmarks being above the helpers
- factoring out shared bench parametrization into `const BENCH_ARGS`, e.g. see `encodings/fsst/benches/fsst_compress.rs`

Not really crucial, more like to get the benchmarks into similar shape and easily human parseable due to consistency.